service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/BaseResourceCreateStep.java

```diff
@@ -144,4 +152,28 @@ private <T extends BaseResourceCreateStep> void registerForNameGeneration(
       ResourceNameProvider resourceNameProvider, T step) {
     resourceNameProvider.registerStep(step);
   }
+
+  private StepResult handleManagementException(
+      ManagementException e,
+      String landingZoneId,
+      String landingZoneType,
+      String managedResourceGroup) {
+    var handled = maybeHandleManagementException(e);
+    if (handled.isPresent()) {
+      var stepResult = handled.get();
+      if (!stepResult.isSuccess()
+          && stepResult.getStepStatus().equals(StepStatus.STEP_RESULT_FAILURE_FATAL)) {
+        MetricUtils.incrementLandingZoneCreationFailure(landingZoneType);
```

Comment:
Same comment, is this indicating creation failure or step failure? Those should be separate metrics.

Reply:
It's indicating step failure which leads to the whole flight failure.